paddle/fluid/pybind/protobuf.cc

```diff
@@ -200,13 +202,19 @@ void BindBlockDesc(py::module &m) {
              return self.FindVarRecursive(name);
            },
            py::return_value_policy::reference)
+      .def("remove_var",
```

Comment:
I began to think our previous design of `XXXDesc` maybe a wrong decision. When we need to manipulate the ProgramDesc proto message, we have to write corresponding c++ implementation, then expose it to pybind, and sync the python holders of the c++ side objects, this makes the code complicated and not simple enough.

When comes with transpilers which need to add, cut, remove operators, variables, and blocks in the ProgramDesc, we must be very careful that the c++ side object reference may have already deleted.

A simpler way is to manipulate the ProgramDesc proto message directly in python, and give it to the c++ executor. For performance, python is slow indeed, but if we only use it in compile time and transpile time, it won't affect the total training performance.

Reply:
@wangkuiyi @reyoung can you please take a look at the above comment?

Reply:
I agree with @typhoonzero. It is so difficult for add, cut, insert, remove operators, variables, and blocks in the ProgramDesc for writing a transpiler, I must change `sync_with_cpp` method in `framework.py` at the same time. 

Related PR: refine sync_with_cpp when remove ops or remove vars https://github.com/PaddlePaddle/Paddle/pull/9600

Reply:
Indeed need some rethink